caip-browser_extension_provider_communication.md

```diff
@@ -0,0 +1,141 @@
+---
+# Every document starts with a front matter in YAML enclosed by triple dashes.
+# See https://jekyllrb.com/docs/front-matter/ to learn more about this concept.
+caip: <to be assigned>
+title: Web extension provider communication
+author: Mark Stacey (@Gudahtt), Jiexi Luan (@jiexi)
+discussions-to: <URL>
+status: Draft
+type: Standard
+created: 2022-11-28
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the CAIP.-->
+The proposal aims to standardize how providers communicate with web extensions, making the process more efficient and secure.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+This CAIP establishes how providers can communicate with web extensions over `externally_connectable` in an extensible manner.
+
+## Motivation
+<!--The motivation is critical for CAIP. It should clearly explain why the state of the art is inadequate to address the problem that the CAIP solves. CAIP submissions without sufficient motivation may be rejected outright.-->
+Web extensions today will typically inject a JavaScript provider API into websites as a global variable. For example, in the Ethereum ecosystem this provider API is standardized in EIP-1193, and is conventionally injected as `window.ethereum`.
+
+This injected API strategy has some advantages:
+* For websites developers, using a global variable is simple and requires no effort on their part to setup.
+* It allows websites to support any web extension following this standard with no additional effort.
+
+However, the injected API strategy has many disadvantages:
+* It depends upon the web extension having read and write access to every website, which is a scary permission that web extension authors might otherwise be able to avoid asking for.
+* It slows down every website by injecting additional code to be parsed and executed. It even slows down the initial page load in most cases, because many web extensions inject the provider synchronously to maintain compatibility with websites that expect it to be available immediately.
+* Injecting code into a website can break it.
+* It provides no way for web extension authors to safely make breaking changes to their provider API, resulting in even more code being injected to support legacy APIs.
```

Comment:
maybe we should explain this one a bit more?

Reply:
tbh, I'm not sure I fully understand it myself haha

Reply:
@Gudahtt

Reply:
I guess I get what its driving at but I'm not clear on how `externally_connectable` resolves the issue of breaking changes other than no longer needing "more code" to be injected

Reply:
Oh I see its answered directly below: https://github.com/jiexi/CAIPs/pull/1/files#diff-c975f3fd9fb25a0964359ba4a49e22edd4fac12cc60d0645ee191f6a02b14069R42

Reply:
reworded here https://github.com/jiexi/CAIPs/pull/1/commits/a2c083a6246d221786366f5cd7d839958bfe04f7